swarm64_tpc_toolkit/correctness.py

```diff
@@ -27,14 +26,22 @@ def has_differences(self, first_df, second_df):
         if first_df.empty != second_df.empty:
             return True
 
-        diff = first_df.merge(second_df, indicator='source', how='outer')
-        self.diff = diff[diff['source'] != 'both']
-        self.diff['source'] = self.diff['source'].apply(
-            lambda x: 'benchmark results' if x == 'left_only' else 'correctness results')
-        diff_rows_count = self.diff.shape[0]
+        first_df = first_df.round(2)
+        second_df = second_df.round(2)
 
-        if diff_rows_count > 0:
-            return True
+        for column in first_df:
+            a = first_df[column]
+            b = second_df[column]
+
+            if a.dtype == 'float64':
+                # absolute(a - b) <= (atol + rtol * absolute(b))
+                # current tolerances: atol=1e-08, rtol=1e-05
+                if not numpy.isclose(a, b).all():
```

Comment:
#current tolerances: atol=1e-08, rtol=1e-05
are those then what the TPC-H spec does? the whole reason for doing it the way i did it is that it follows the spec very precisely.